mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/scaladsl/MqttSource.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming.scaladsl
+
+import akka.Done
+import akka.annotation.{ApiMayChange, InternalApi}
+import akka.stream.alpakka.mqtt.streaming._
+import akka.stream.alpakka.mqtt.streaming.impl.HighLevelMqttSource
+import akka.stream.scaladsl.Source
+
+import scala.collection.immutable
+import scala.concurrent.Future
+
+@ApiMayChange
+trait MqttAckHandle {
+
+  /** Acknowledge received data. */
+  def ack(): Future[Done]
+}
+
+/**
+ * Scala API
+ */
+@ApiMayChange
+object MqttSource {
+
+  /**
+   * High-level API to subscribe to MQTT topics with at-most-once semantics.
+   */
+  @ApiMayChange
+  def atMostOnce(
+      mqttClientSession: MqttClientSession,
```

Comment:
Given that you're controlling the entire connection, you can also control the session and not require the user to pass it in.

Reply:
Ok.